msdsales/read.go

```diff
@@ -0,0 +1,95 @@
+package msdsales
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"net/url"
+	"strings"
+
+	"github.com/amp-labs/connectors/common"
+)
+
+// TODO MS Sales allows fine control of reading
+// TODO basic read
+// $select = ReadParams{fields} [COMPLETED]
+// $expand = nested response
+// $orderby = list of fields with asc/desc keyword
+// TODO batch
+// $apply = batch
+// TODO search
+// $filter = query functions, comparisons
+// TODO pagination
+// $top = <int> of entries to return (ignored if header <Prefer: odata.maxpagesize>)
+// $count = counts all existing rows (@odata.count)
+
+var annotationsHeader = common.Header{
+	Key:   "Prefer",
+	Value: "odata.include-annotations=\"*\"", // TODO we can specify which annotations to include
+}
+
+func newPaginationHeader(pageSize int) common.Header {
+	return common.Header{
+		Key:   "Prefer",
+		Value: fmt.Sprintf("odata.maxpagesize=%v", pageSize),
+	}
+}
+
+func (c *Connector) Read(ctx context.Context, config common.ReadParams) (*common.ReadResult, error) {
+	var fullURL string
+
+	if len(config.NextPage) == 0 {
+		// First page
+		relativeURL := config.ObjectName + makeQueryValues(config)
+		fullURL = c.getURL(relativeURL)
+	} else {
+		// Next page
+		fullURL = config.NextPage
+	}
+	// TODO given that one of the fields is annotation we can automatically add annotation header (how the hell the end user gonna know about the names of those fields)
+	rsp, err := c.get(ctx, fullURL, newPaginationHeader(resolvePageSize(config)), annotationsHeader)
```

Comment:
Response can be more descriptive if annotation header is provided.
Right now it is always passed. How much control will we give? It relates to `$expand` query for descriptive responses.